src/main/java/com/google/starfish/servlets/SearchServlet.java

```diff
@@ -0,0 +1,94 @@
+package com.google.starfish.servlets;  
+
+import java.io.IOException;  
+import java.sql.Connection;  
+import java.sql.SQLException;  
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import javax.servlet.ServletException;  
+import javax.servlet.annotation.WebServlet;  
+import javax.servlet.http.HttpServlet;  
+import javax.servlet.http.HttpServletRequest;  
+import javax.servlet.http.HttpServletResponse;  
+import javax.sql.DataSource;
+import java.util.Date;
+import com.google.starfish.services.NoteService;
+import com.google.starfish.models.Note;
+import com.google.gson.Gson;
+import java.util.ArrayList;
+
+/** Servlet that returns search results for notes based on school and course */
+@WebServlet("/search")  
+public class SearchServlet extends HttpServlet {  
+
+  private NoteService noteService = new NoteService();
+
+  @Override
+  public void doGet(HttpServletRequest req, HttpServletResponse res) throws ServletException, IOException {
+    DataSource pool = (DataSource) req.getServletContext().getAttribute("my-pool");  
+    String reqSchool = req.getParameter("school").toLowerCase().trim();
+    String reqCourse= req.getParameter("course").toLowerCase().trim();
+    ArrayList<Note> notes = new ArrayList<>();
+    try (Connection conn = pool.getConnection()) {
+      String stmt = getSearchQuery(reqSchool, reqCourse);
+      try (PreparedStatement getNotesStatement = conn.prepareStatement(stmt)) {
+        getNotesStatement.setString(1, reqSchool);
+        getNotesStatement.setString(2, reqCourse);
+        ResultSet rs = getNotesStatement.executeQuery();
+
+        while (rs.next()) {
+          long noteId = rs.getLong("id");
+          long authorId = rs.getLong("author_id");
+          String school = rs.getString("school");
+          String course = rs.getString("course");
+          String title = rs.getString("title");
+          String sourceUrl = rs.getString("title");
+          String pdfSource = rs.getString("pdf_source");
+          Date dateCreated = rs.getDate("date_created");
+          long numDownloads = rs.getLong("num_downloads");
+          long numFavorites = noteService.getNumFavoritesById(pool, noteId);
+
+          Note thisNote = new Note.Builder()
+                              .setId(noteId)
+                              .setAuthorId(authorId)
+                              .setRequiredLabels(school, course)
+                              .setNoteTitle(title)
+                              .setSourceUrl(sourceUrl)
+                              .setOptionalPdfSource(pdfSource)
```

Comment:
I know we talked about why we need both of these fields, but right now the frontend only accepts a single pdfSrc field in the Note object since we haven't implemented anything other than pdf storage.

Reply:
Sounds good, then I think the Note objects returned to client that just have a pdfSource would hold that in both the sourceUrl and the "optionalPdfSource". This would just be the blobkey, and once we get source URLs for google docs, for any Notes that actually have an associated Google Docs, that's the URL we'd keep in sourceUrl